iceoryx_hoofs/reporting/include/iox/assertions.hpp

```diff
@@ -44,19 +44,29 @@
         iox::er::forwardPanic(CURRENT_SOURCE_LOCATION, message);                                                       \
     } while (false)
 
-/// @brief report fatal error if expr evaluates to false
-/// @note for conditions that may actually happen during correct use
-/// @param expr boolean expression that must hold
-/// @param error error object (or code)
-#define IOX_REQUIRE(expr, error) IOX_REPORT_FATAL_IF(!(expr), error)
-
 //************************************************************************************************
 //* For documentation of intent, defensive programming and debugging
 //*
 //* There are no error codes/errors required here on purpose, as it would make the use cumbersome.
 //* Instead a special internal error type is used.
 //************************************************************************************************
 
+/// @brief report fatal required condition violation if expr evaluates to false
+/// @note for conditions that may actually happen during correct use
+/// @param expr boolean expression that must hold
+/// @param message message to be forwarded in case of violation
+#define IOX_REQUIRE(expr, message)                                                                                     \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        if (!(expr))                                                                                                   \
+        {                                                                                                              \
+            iox::er::forwardFatalError(iox::er::Violation::createRequiredConditionViolation(),                         \
+                                       iox::er::REQUIRED_CONDITION_VIOLATION,                                          \
+                                       CURRENT_SOURCE_LOCATION,                                                        \
+                                       message); /* @todo iox-#1032 add strigified 'expr' as '#expr' */                \
+        }                                                                                                              \
+    } while (false)
```

Comment:
@elBoberido I recently considered as trailing `static_assert` as an alternative to this `do { ... } while` madness: https://stackoverflow.com/questions/35530850/how-to-require-a-semicolon-after-a-macro.

However, it is not equivalent in general as it e.g. does not play nice in `if(...) MACRO() else { }` without parentheses. One could argue this is a good thing but I disagree as the MACRO should not impose these restrictions if possible. Also there may be more limitations. 

In other words I have no better alternative yet and this is pretty standard but also ancient from the C world.

Stringification as in the TODO is a good addition.